spark/src/main/java/org/opensearch/sql/spark/functions/response/DefaultSparkSqlFunctionResponseHandle.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.functions.response;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.opensearch.sql.data.model.ExprBooleanValue;
+import org.opensearch.sql.data.model.ExprByteValue;
+import org.opensearch.sql.data.model.ExprDateValue;
+import org.opensearch.sql.data.model.ExprDoubleValue;
+import org.opensearch.sql.data.model.ExprFloatValue;
+import org.opensearch.sql.data.model.ExprIntegerValue;
+import org.opensearch.sql.data.model.ExprLongValue;
+import org.opensearch.sql.data.model.ExprShortValue;
+import org.opensearch.sql.data.model.ExprStringValue;
+import org.opensearch.sql.data.model.ExprTimestampValue;
+import org.opensearch.sql.data.model.ExprTupleValue;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.data.type.ExprCoreType;
+import org.opensearch.sql.data.type.ExprType;
+import org.opensearch.sql.executor.ExecutionEngine;
+
+/**
+ * Default implementation of SparkSqlFunctionResponseHandle.
+ */
+public class DefaultSparkSqlFunctionResponseHandle implements SparkSqlFunctionResponseHandle {
+
+  private final JSONObject responseObject;
+  private Iterator<ExprValue> responseIterator;
+  private ExecutionEngine.Schema schema;
+
+  /**
+   * Constructor.
+   *
+   * @param responseObject Spark responseObject.
+   */
+  public DefaultSparkSqlFunctionResponseHandle(JSONObject responseObject) {
+    this.responseObject = responseObject;
+    constructIteratorAndSchema();
+  }
+
+  private void constructIteratorAndSchema() {
+    List<ExprValue> result = new ArrayList<>();
+    List<ExecutionEngine.Schema.Column> columnList;
+    if (responseObject.has("data")) {
+      JSONObject items = responseObject.getJSONObject("data");
+      columnList = getColumnList(items.getJSONArray("schema"));
+      for (int i = 0; i < items.getJSONArray("result").length(); i++) {
+        JSONObject row = new JSONObject(
+            items.getJSONArray("result").get(i).toString().replace("'", "\""));
+        LinkedHashMap<String, ExprValue> linkedHashMap = extractRow(row, columnList);
+        result.add(new ExprTupleValue(linkedHashMap));
+      }
+    } else {
+      throw new RuntimeException("Unexpected result during spark sql query execution");
```

Comment:
No. it should still have stepId & applicationId information. This check was just precautionary in case something went wrong with json object. can be removed. 
Also tested create query and received empty response. 
```
{
    "schema": [],
    "datarows": [],
    "total": 0,
    "size": 0
}
```